app/src/main/java/com/conupods/Settings/SettingsActivity.java

```diff
@@ -0,0 +1,112 @@
+package com.conupods.Settings;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.content.SharedPreferences.Editor;
+import android.os.Bundle;
+import android.widget.Button;
+import android.widget.CheckBox;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.conupods.OutdoorMaps.View.MapsActivity;
+import com.conupods.R;
+
+public class SettingsActivity extends AppCompatActivity {
```

Comment:
`AppCompatActivity` is the reason you access to getApplicationContext so I am not sure what the best practice is anymore...maybe you dont need to change anything